src/app/routes/edit/edit.component.ts

```diff
@@ -32,7 +53,145 @@ export class EditComponent implements OnInit {
+      }
+
+      // Show the popup
+      function showPopup () {
+        // Get the popup object defined above.
+        var $popup = editor.popups.get('customPlugin.popup');
+
+        // If popup doesn't exist then create it.
+        // To improve performance it is best to create the popup when it is first needed
+        // and not when the editor is initialized.
+        if (!$popup) $popup = initPopup();
+
+        // Set the editor toolbar as the popup's container.
+        editor.popups.setContainer('customPlugin.popup', editor.$tb);
+
+        // This custom popup is opened by pressing a button from the editor's toolbar.
+        // Get the button's object in order to place the popup relative to it.
+        var $btn = editor.$tb.find('.fr-command[data-cmd="insert_profile"]');
+
+        // Compute the popup's position.
+        var left = $btn.offset().left + $btn.outerWidth() / 2;
+        var top = $btn.offset().top + (editor.opts.toolbarBottom ? 10 : $btn.outerHeight() - 10);
+
+        // Show the custom popup.
+        // The button's outerHeight is required in case the popup needs to be displayed above it.
+        editor.popups.show('customPlugin.popup', left, top, $btn.outerHeight());
+      }
+
+      // Hide the custom popup.
+      function hidePopup () {
+        editor.popups.hide('customPlugin.popup');
+      }
+
+      // Methods visible outside the plugin.
+      return {
+        showPopup: showPopup,
+        hidePopup: hidePopup
+      }
+    }
+
+    // Define an icon and command for the button that opens the custom popup.
+    $.FroalaEditor.DefineIcon('insert_profile', { NAME: 'plus'})
+    $.FroalaEditor.RegisterCommand('insert_profile', {
+      title: 'Insert Profile',
+      undo: false,
+      focus: false,
+      popup: true,
+      // Buttons which are included in the editor toolbar should have the plugin property set.
+      plugin: 'customPlugin',
+      callback: function () {
+        if (!this.popups.isVisible('customPlugin.popup')) {
+          this.customPlugin.showPopup();
+        }
+        else {
+          if (this.$el.find('.fr-marker')) {
+            this.events.disableBlur();
+            this.selection.restore();
+          }
+          this.popups.hide('customPlugin.popup');
+        }
+      }
+    });
+
+    // Define custom popup close button icon and command.
+    $.FroalaEditor.DefineIcon('popupClose', { NAME: 'times' });
+    $.FroalaEditor.RegisterCommand('popupClose', {
+      title: 'Close',
+      undo: false,
+      focus: false,
+      callback: function () {
+        this.customPlugin.hidePopup();
+      }
+    });
+
+    // Define custom popup 1.
+    $.FroalaEditor.DefineIcon('insertProfileButton', { NAME: 'plus' });
+    $.FroalaEditor.RegisterCommand('insertProfileButton', {
+      title: 'Insert Profile',
+      undo: false,
+      focus: false,
+      callback: function () {
+        $.get("http://localhost:8888/profile/1718/" + $('#usr').val(), (data, status) => {
```

Comment:
I just realized we could use [`localStorage` ](https://developer.mozilla.org/en-US/docs/Web/API/Storage/LocalStorage)  as a way to set the `CURRENTY_YEAR` and `environment.SERVER_URL` globally.